fulp_modules/bloodsuckers/bloodsucker/bloodsucker_coffin.dm

```diff
@@ -0,0 +1,176 @@
+			if (bloodsuckerdatum && bloodsuckerdatum.coffin == src)
+				bloodsuckerdatum.coffin = null
+				bloodsuckerdatum.lair = null
+			to_chat(resident, "<span class='danger'><span class='italics'>You sense that the link with your coffin, your sacred place of rest, has been broken! You will need to seek another.</span></span>")
+		resident = null // Remove resident. Because this object isnt removed from the game immediately (GC?) we need to give them a way to see they don't have a home anymore.
+
+/obj/structure/closet/crate/coffin/can_open(mob/living/user)
+	// You cannot lock in/out a coffin's owner. SORRY.
+	if (locked)
+		if(user == resident)
+			if (welded)
+				welded = FALSE
+				update_icon()
+			//to_chat(user, "<span class='notice'>You flip a secret latch and unlock [src].</span>") // Don't bother. We know it's unlocked.
+			locked = FALSE
+			return 1
+		else
+			playsound(get_turf(src), 'sound/machines/door_locked.ogg', 20, 1)
+			to_chat(user, "<span class='notice'>[src] is locked tight from the inside.</span>")
+	return ..()
+
+/obj/structure/closet/crate/coffin/close(mob/living/user)
+	if (!..())
+		return FALSE
+	// Only the User can put themself into Torpor. If already in it, you'll start to heal.
+	if ((user in src))
+		var/datum/antagonist/bloodsucker/bloodsuckerdatum = user.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+		if (bloodsuckerdatum)
+			LockMe(user)
+			if (!bloodsuckerdatum.coffin && !resident) //Claim?
+				switch(alert(user,"Do you wish to claim this as your coffin? [get_area(src)] will be your lair.","Claim Lair","Yes", "No"))
+					if("Yes")
+						ClaimCoffin(user)
+			if(user.AmStaked()) //Staked? Dont heal
+				to_chat(bloodsuckerdatum.owner.current, "<span class='userdanger'>You are staked! Remove the offending weapon from your heart before sleeping.</span>")
+				return
+			// Heal
+			if(bloodsuckerdatum.HandleHealing(0)) // Healing Mult 0 <--- We only want to check if healing is valid!
+				to_chat(bloodsuckerdatum.owner.current, "<span class='notice'>You enter the horrible slumber of deathless Torpor. You will heal until you are renewed.</span>")
+				bloodsuckerdatum.Torpor_Begin()
+			bloodsuckerdatum.SpendRank() // Level up? Auto-Fails if not appropriate
+	return TRUE
+
+/obj/structure/closet/crate/coffin/attackby(obj/item/W, mob/user, params)
+	// You cannot weld or deconstruct an owned coffin.
+	if (resident != null && user != resident) // Owner can destroy their own coffin.
+		if(opened)
+			if(istype(W, cutting_tool))
+				to_chat(user, "<span class='notice'>This is a much more complex mechanical structure than you thought. You don't know where to begin cutting [src].</span>")
+				return
+		else if(anchored && istype(W, /obj/item/wrench)) // Can't unanchor unless owner.
+			to_chat(user, "<span class='danger'>The coffin won't come unanchored from the floor.</span>")
+			return
+
+	if(locked && istype(W, /obj/item/crowbar))
```

Comment:
so you can pry the lid off of a locked cargo crate with a crowbar if it's a vampire's home coffin, but not if it's just a normal (locked) crate?

Reply:
also, uh, pretty much the same thing I said about wrenches and welders applies here